6-AdvancedScenarios/4-sign-in-hybrid/README.md

```diff
@@ -0,0 +1,357 @@
+    * Select optional claim name, choose **login_hint**.
+1. Select **Add** to save changes.
+
+#### Configure the app (msal-hybrid-spa) to use your app registration
+
+Open the project in your IDE (like Visual Studio or Visual Studio Code) to configure the code.
+
+> In the steps below, "ClientID" is the same as "Application ID" or "AppId".
+
+1. Open the `App/.env` file.
+1. Find the key `Enter_the_Application_Id_Here` and replace the existing value with the application ID (clientId) of `msal-hybrid-spa` app copied from the Azure portal.
+1. Find the key `Enter_the_Tenant_Info_Here` and replace the existing value with your Azure AD tenant ID.
+1. Find the key `Enter_the_Client_Secret_Here` and replace the existing value with the key you saved during the creation of `msal-hybrid-spa` copied from the Azure portal.
+
+In addition:
+
+1. Open the `App/client/authConfig.js` file.
+1. Find the string `Enter_the_Application_Id_Here` and replace the existing value with the application ID (clientId) of `msal-hybrid-spa` app copied
+1. Find the string `Enter_the_Tenant_Info_Here` and replace the existing value with your Azure AD tenant ID.
+
+### Step 4: Running the sample
+
+For command line run the next commands:
+
+* Run the app:
+
+  ```console
+    cd 6-AdvancedScenarios\4-sign-in-hybrid\App
+    npm start
+    ```
+
+## About the code
+
+### Confidential client
+
+In this sample, the user is first authenticated to backend using an MSAL Node confidential client application.
+
+```javascript
+  const msal = require('@azure/msal-node');
+  require('dotenv').config();
+
+  const msalInstance = new msal.ConfidentialClientApplication({
+      auth: {
+          clientId: process.env.CLIENT_ID,
+          authority: `https://login.microsoftonline.com/${process.env.TENANT_ID}`,
+          clientSecret: process.env.CLIENT_SECRET
+      },
+      system: {
+          loggerOptions: {
+              loggerCallback: (loglevel, message, containsPii) => {
+                  console.log(message);
+              },
+              piiLoggingEnabled: false,
+              logLevel: msal.LogLevel.Verbose,
+          }
+      }
+  });
+
+```
+
+Begin by generating an authorization code URL and redirect the app to it:
+
+```javascript
+ const authCodeUrlParameters = {
+        redirectUri: process.env.REDIRECT_URI,
+        responseMode: "form_post",
+    };
+
+    msalInstance.getAuthCodeUrl(authCodeUrlParameters)
+        .then((response) => {
+            res.json(response);
+        }).catch((error) => console.log(error))
+```
+
+Next, parse the authorization code, and invoke the `acquireTokenByCode` API by setting the `enableSpaAuthorizationCode` to true,  which will enable MSAL to acquire a second authorization code to be redeemed by your single-page application. Your application should parse this second authorization code, as well as any account hints (e.g. sid, login_hint, preferred_username) and return them such that they can be obtained client-side:
```

Comment:
> Your application should parse this second authorization code [](http://example.com/codeflow?start=239&length=61)

really ?

Reply:
@salman90 it would be better to say something like "get the auth code from request body" or something

Reply:
Updated!